File: app/src/main/java/dev/blackcat/minauta/sync/AvailableTimeThread.java
package dev.blackcat.minauta.sync;

import dev.blackcat.jnauta.net.Authentication;
import dev.blackcat.jnauta.net.ConnectionBuilder;
import dev.blackcat.minauta.data.Account;

public class AvailableTimeThread extends SyncRunnable<String>
{

    Account account;

    public AvailableTimeThread(Account account)
    {
        this.account = account;
    }

    @Override
    public void run()
    {
        Authentication authentication = new Authentication(ConnectionBuilder.Method.OK_HTTP, account.getUsername(), account.getPassword(), null);
        String r = authentication.getAvailableTime(account.getSession().getTimeParams());
        this.setResult(r);
    }
}

File: app/src/main/java/dev/blackcat/minauta/sync/AvailableTimeThread.kt
package dev.blackcat.minauta.sync

import dev.blackcat.jnauta.net.Authentication
import dev.blackcat.jnauta.net.ConnectionBuilder
import dev.blackcat.minauta.data.Account

class AvailableTimeThread(internal var account: Account) : SyncRunnable<String>() {

    override fun run() {
        val authentication = Authentication(ConnectionBuilder.Method.OK_HTTP, account.username, account.password, null)
        val r = authentication.getAvailableTime(account.session!!.loginParams)
        this.result = r
    }
}